lib/src/parsers/phone_parser.dart

```diff
@@ -128,14 +128,47 @@ class PhoneParser {
   ///
   /// throws a PhoneNumberException if the country calling code is invalid
   @internal
-  static PhoneNumber fromRaw(String phoneNumber) {
+  static PhoneNumber fromRaw(String phoneNumber, { IsoCode? callerCountry, IsoCode? destinationCountry }) {
     phoneNumber = TextParser.normalize(phoneNumber);
-    phoneNumber = InternationalPrefixParser.removeInternationalPrefix(
-      phoneNumber,
-    );
-    final countryCode = CountryCodeParser.extractCountryCode(phoneNumber);
+    String withoutIntlPrefix; // this shall contain the number without exit codes, starting with the country code. e.g. 49301234567
+    if (callerCountry != null) {
+      // as we know the caller country, we can remove the possible international prefix (exit code)
+      var metaCallerCountry = MetadataFinder.getMetadataForIsoCode(callerCountry);
+      withoutIntlPrefix =
+          InternationalPrefixParser.removeInternationalPrefix(
+              phoneNumber, metadata: metaCallerCountry);
+      if (withoutIntlPrefix.length == phoneNumber.length) {
+        // no exit code was removed, so the given number was not an international number
+        // since we know the caller country we may assemble the number with leading country code for further processing
+        var nationalNumber = NationalNumberParser.removeNationalPrefix(withoutIntlPrefix, metaCallerCountry);
+        withoutIntlPrefix = metaCallerCountry.countryCode + nationalNumber;
+        // no need to extract the destination country then, it must be the same as the caller country
+        destinationCountry ??= callerCountry;
+      }
+    } else {
+      // we don't know the caller country, so we just make a best guess
+      withoutIntlPrefix  =
+          InternationalPrefixParser.removeInternationalPrefix(
+              phoneNumber);
+    }
+    // withoutIntlPrefix now starts with a country code, regardless of the number format initially given
+    late String countryCode;
+    if (destinationCountry != null) {
+      // if we are given a destination country we check if the country code in the phone number matches the destination country code
+      var metaDestinationCountry = MetadataFinder.getMetadataForIsoCode(destinationCountry);
+      if (metaDestinationCountry.countryCode != withoutIntlPrefix.substring(0,metaDestinationCountry.countryCode.length)) {
+        throw PhoneNumberException(
+            code: Code.invalidCountryCallingCode,
+            description: "The country code in the given phone number does not match the expected destination country");

```

Comment:
There is still something wrong here though.

Given: "479 22 22 22", dest : BE, caller: Germany. This will throw. Imo the 49 should not be added line 143 if the dest is provided